payment/datastore.go

```diff
@@ -403,6 +406,77 @@ func (pg *Postgres) IsStripeSub(orderID uuid.UUID) (bool, string, error) {
 	return ok, "", err
 }
 
+// RenewOrder updates the orders status to paid and paid at time, inserts record of this order
+// 	Status should either be one of pending, paid, fulfilled, or canceled.
+func (pg *Postgres) RenewOrder(ctx context.Context, orderID uuid.UUID) error {
+	// create tx
+	tx, err := pg.RawDB().BeginTxx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer pg.RollbackTx(tx)
+
+	// how long should the order be valid for?
+	var orderTimeBounds = struct {
+		ValidFor *time.Duration `db:"valid_for"`
+		LastPaid sql.NullTime   `db:"last_paid"`
+	}{}
+
+	err = tx.GetContext(ctx, &orderTimeBounds, `
+		SELECT valid_for, last_paid
+		FROM orders
+		WHERE id = $1
+	`, orderID)
+	if err != nil {
+		return fmt.Errorf("unable to get order time bounds: %w", err)
+	}
+
+	// update last paid
+	lastPaid := time.Now()
+	var expiresAt time.Time
+	if orderTimeBounds.ValidFor != nil && orderTimeBounds.LastPaid.Valid {
+		expiresAt = orderTimeBounds.LastPaid.Time.Add(*orderTimeBounds.ValidFor)
+	}
+
+	result, err := tx.ExecContext(ctx, `
+		UPDATE orders
+		SET
+			status = 'paid',
+			updated_at = CURRENT_TIMESTAMP,
+			last_paid_at = $1,
+			expires_at = $2
+		WHERE 
+			id = $3
+	`, lastPaid, expiresAt, orderID)
+
+	if err != nil {
+		return err
+	}
+
+	rowsAffected, err := result.RowsAffected()
+	if rowsAffected == 0 || err != nil {
+		return errors.New("no rows updated")
+	}
+
+	result, err = tx.ExecContext(ctx, `
+		INSERT INTO order_renewal
```

Comment:
would we miss recording initial payment in this table and lose history since this is only triggered for renewal?

Reply:
yes, this updates UpdateOrder, CreateOrder with recording on OrderStatusPaid now. https://github.com/brave-intl/bat-go/pull/894/commits/c600e68352dea35c44e26d10084bd27dedff2cbf

Reply:
I guess the confusing part of this table naming is that `renewal` implies restarting a subscription, but it seems like we use `order_renewal` for both new subscriptions and renewing subscriptions?

Reply:
`order_payment_history`? :P

Reply:
I think we can preserve history if we rename `order_renewal` to `order_subscriptions` with a start-and-end-time and move the `expires_at` timestamp to the `order_subscriptions` table.  I still think the `valid_for` isn't elegant, but we can say `requested_subscription_duration` instead?

The `status` field on `orders` should swap `paid` for `active`, since either there is an active credit card & the user chooses to continue giving money at the end of the last `order_subscriptions`.

Reply:
I suggest maybe we make an issue to redesign this a bit to handle subscriptions better.  There are a lot of improvements that could be done.  For color, valid_for is being used as the sku structure has a "credential valid duration" which is why we are using that at the moment.

Reply:
https://github.com/brave-intl/bat-go/pull/894/commits/10be0fd563790266a216d97e70d87389d4b146d4 for order_payment_history for the time being and we can address a subscription redesign in an issue.